libs/platform/unit/platform_unit_test.cpp

```diff
@@ -371,3 +375,82 @@ TEST_CASE("serialize_map_test", "[platform]")
+    ebpf_program_type_descriptor_t program_type = {"unit_test_program", &context_descriptor, PROGRAM_TYPE_TEST};
+    ebpf_program_information_t in_program_info = {program_type, EBPF_COUNT_OF(helper_prototype), helper_prototype};
+
+    size_t buffer_length = 0;
+    uint8_t* buffer = nullptr;
+    size_t required_length;
+    size_t serialized_length;
+
+    ebpf_program_information_t* out_program_info;
+
+    // Serialize.
+    ebpf_result_t result = ebpf_serialize_program_information(
+        &in_program_info, buffer, buffer_length, &serialized_length, &required_length);
+    REQUIRE(result == EBPF_INSUFFICIENT_BUFFER);
+
+    buffer = static_cast<uint8_t*>(calloc(required_length, 1));
+    buffer_length = required_length;
+
+    result = ebpf_serialize_program_information(
+        &in_program_info, buffer, buffer_length, &serialized_length, &required_length);
+    REQUIRE(result == EBPF_SUCCESS);
+
+    // Deserialize.
+    result = ebpf_deserialize_program_information(serialized_length, buffer, &out_program_info);
+    REQUIRE(result == EBPF_SUCCESS);
+
+    // Verify de-serialized program information matches input.
+    REQUIRE(
+        in_program_info.program_type_descriptor.platform_specific_data ==
+        out_program_info->program_type_descriptor.platform_specific_data);
+    REQUIRE(
+        in_program_info.program_type_descriptor.is_privileged ==
+        out_program_info->program_type_descriptor.is_privileged);
+    REQUIRE(in_program_info.program_type_descriptor.context_descriptor != nullptr);
+    REQUIRE(
+        memcmp(
+            in_program_info.program_type_descriptor.context_descriptor,
+            out_program_info->program_type_descriptor.context_descriptor,
+            sizeof(ebpf_context_descriptor_t)) == 0);
+    REQUIRE(
+        strncmp(
+            in_program_info.program_type_descriptor.name,
+            out_program_info->program_type_descriptor.name,
+            EBPF_MAX_PROGRAM_DESCRIPTOR_NAME_LENGTH) == 0);
+    REQUIRE(in_program_info.count_of_helpers == out_program_info->count_of_helpers);
+    REQUIRE(out_program_info->helper_prototype != nullptr);
+    for (uint32_t i = 0; i < in_program_info.count_of_helpers; i++) {
+        ebpf_helper_function_prototype_t* in_prototype = &in_program_info.helper_prototype[i];
+        ebpf_helper_function_prototype_t* out_prototype = &out_program_info->helper_prototype[i];
+        REQUIRE(in_prototype->helper_id == out_prototype->helper_id);
+        REQUIRE(in_prototype->return_type == out_prototype->return_type);
+        for (int j = 0; j < 5; j++)
+            REQUIRE(in_prototype->arguments[j] == out_prototype->arguments[j]);
+        REQUIRE(out_prototype->name != nullptr);
+        REQUIRE(strncmp(in_prototype->name, out_prototype->name, EBPF_MAX_HELPER_FUNCTION_NAME_LENGTH) == 0);
+    }
+
+    // Free de-serialized program information.
+    ebpf_program_information_free(out_program_info);
+
+    if (buffer != nullptr)
+        free(buffer);
```

Comment:
```suggestion
    free(buffer);
```
reduce code complexity... free() is always safe to call with nullptr